openff/toolkit/tests/test_forcefield.py

```diff
@@ -3383,6 +3383,37 @@ def test_charges_on_ref_mols_when_using_return_topology(
                 all_charges_zero = False
         assert not (all_charges_zero)
 
+    def test_library_charges_from_molecule(self):
+        """Test that constructing a LibraryChargeHandler from partial charges on Molecule objects
+        produces the same result as using the `charge_from_molecules` kwarg."""
+        # TODO: Remove this test if `charge_from_molecules` is depcreated (#806)
+        mol = Molecule.from_mapped_smiles(
+            "[C:1]([C:2]([O:3][H:9])([H:7])[H:8])([H:4])([H:5])[H:6]"
+        )
+        mol.partial_charges = np.linspace(-0.4, 0.4, 9) * unit.elementary_charge
```

Comment:
This test is failing because some of the hydrogens in ethanol are chemically indistinguishable, but the test expects them to get different partial charges. To get a molecule with all-unique atoms, maybe try

```suggestion
        mol = Molecule.from_mapped_smiles(
            "[Cl:1]/[C:2]([H:3])=[C:4]([H:5])/[F:6]]"
        )
        mol.partial_charges = np.linspace(-0.25, 0.25, 6) * unit.elementary_charge
```

Reply:
This solution works, but does it pose a general problem to this method? The problem I'm hung up on is if users bring molecules with different partial charges on chemically similar atoms and are surprised by this sort of result.

Reply:
It's not a problem in that it's the intended behavior of the toolkit (chemical perception-based parameter assignment --> a FF should not do different things with chemically indistinguishable atoms) , but it is a problem in that it will surprise users. So it's most likely a documentation problem.